### PR TITLE
Improve build

### PR DESCRIPTION
**Pull request by Lctrs**

### CODE REVIEW

**Comment by Lctrs** [comment]
Here is the diff for the available modules between the one available in the README and the one from the most recent build : 

```
 [PHP Modules]
 bcmath
+bz2
 calendar
 Core
 ctype
 curl
 date
-dba
 dom
 exif
 fileinfo
 filter
 ftp
 gd
+gettext
 hash
 iconv
 intl
 json
 libxml
 mbstring
+mysqli
 mysqlnd
 openssl
 pcntl
 pcre
 PDO
+pdo_mysql
 pdo_sqlite
 Phar
 posix
+pspell
+readline
 Reflection
 session
 SimpleXML
 sockets
 SPL
 sqlite3
 standard
 sysvmsg
+sysvsem
 sysvshm
 tokenizer
 wddx
 xdebug
 xml
 xmlreader
+xmlrpc
 xmlwriter
+xsl
 zip
 zlib
 
 [Zend Modules]
 Xdebug
```
`dba` not being available anymore is expected as it's built as a shared module.

~~We should bundle the `re2c` library. PHP use it as a lexer generator.~~ Only needed if we plan to do modifications to PHP Core. Could be useful but it's not urgent IMHO.

Some extensions are commented out because their build requires some missing deps : 
- for `enchant`, `gmp`, `imap`, `recode`, `snmp`, `tidy` : based on my experience as PHP developer, remove them. They're not often used.
- for `ldap`, `password-argon2`, `pdo-pgsql`, `sodium` : bundle the missing deps and include them.

Ping @p1u3o, wdyt ?

**Comment by stobbsm** [comment]
Thanks for jumping in and adding all this. Building PHP was new to me, so I only enabled what I needed.

**Comment by p1u3o** [comment]
Seems fine to me, the main usage of this is so IDEs can run their PHP language servers.

File: org.freedesktop.Sdk.Extension.php73.json [hunk 1]
```diff
@@ -22,9 +22,56 @@
...
+			],
+			"sources": [
+				{
+					"type": "archive",
+					"url": "https://www.openldap.org/software/download/OpenLDAP/openldap-release/openldap-2.4.48.tgz",
+					"sha256": "d9523ffcab5cd14b709fcf3cb4d04e8bc76bb8970113255f372bc74954c6074d"
+				},
+				{
+					"type": "patch",
+					"path": "openldap-aarch64.patch"
```
Comment (p1u3o) [inline org.freedesktop.Sdk.Extension.php73.json]: If this is only useful for aarch64, you can use         `"only-arches": "aarch64"` here

**Comment by Lctrs** [comment]
@p1u3o @stobbsm I consider this ready for review now.

**Review by stobbsm — APPROVED** [review]
Test build worked as expected for me. Thanks for all the hard work!

**Comment by stobbsm** [comment]
Anymore changes you want to get in before I merge?

**Comment by Lctrs** [comment]
Nope. Good for me.

**Comment by Lctrs** [comment]
Rebased.

File: org.freedesktop.Sdk.Extension.php73.json [hunk 2]
```diff
@@ -19,12 +35,67 @@
...
+			"sources": [
+				{
+					"type": "archive",
+					"url": "https://www.openldap.org/software/download/OpenLDAP/openldap-release/openldap-2.4.48.tgz",
+					"sha256": "d9523ffcab5cd14b709fcf3cb4d04e8bc76bb8970113255f372bc74954c6074d"
+				},
+				{
+					"type": "patch",
+					"path": "openldap-aarch64.patch",
+                    "only-arches": [
```
Comment (p1u3o) [inline org.freedesktop.Sdk.Extension.php73.json]: White space here is off.

Comment (Lctrs) [reply org.freedesktop.Sdk.Extension.php73.json]: Fixed.

Comment (p1u3o) [reply org.freedesktop.Sdk.Extension.php73.json]: Thanks

File: org.freedesktop.Sdk.Extension.php73.json [hunk 3]
```diff
@@ -19,12 +35,67 @@
...
+		{
+			"name": "libpq",
+			"buildsystem": "simple",
+			"build-commands": [
+				"./configure --prefix=${FLATPAK_DEST} --disable-rpath --with-openssl --without-readline",
+				"cd src/interfaces/libpq",
+				"make -j$FLATPAK_BUILDER_N_JOBS",
+				"make install"
+			],
+			"sources":[
```
Comment (p1u3o) [inline org.freedesktop.Sdk.Extension.php73.json:80]: Lack of space here before the `[`

Just a nit pick, but you have done a space before for the other ones.

File: org.freedesktop.Sdk.Extension.php73.json [hunk 4]
```diff
@@ -132,8 +210,8 @@
 				"--with-php-config=/usr/lib/sdk/php73/bin/php-config"
 			],
 			"post-install": [
-				"install -Dm755 modules/xdebug.so ${FLATPAK_DEST}/lib/php/extensions/no-debug-non-zts-20180731/xdebug.so",
-				"install -Dm644 xdebug.ini ${FLATPAK_DEST}/etc/php/xdebug.ini",
+				"install -Dm755 modules/xdebug.so $(php -r 'echo ini_get(\"extension_dir\");')/xdebug.so",
+				"install -Dm644 xdebug.ini ${PHP_INI_DIR}/conf.d/xdebug.ini",
 				"install -Dm644 LICENSE ${FLATPAK_DEST}/share/licenses/xdebug/LICENSE"
 			],
 			"sources":[
```
Comment (p1u3o) [inline org.freedesktop.Sdk.Extension.php73.json:217]: Here too.

File: org.freedesktop.Sdk.Extension.php73.json (hunk 3 above)
Comment (Lctrs) [reply org.freedesktop.Sdk.Extension.php73.json:80]: Yeah but existing code doesn't use a space after `:`.

Maybe we should standardize all of this in another PR ? This one is already huge.

Comment (p1u3o) [reply org.freedesktop.Sdk.Extension.php73.json:80]: Seems good with me.

**Comment by p1u3o** [comment]
This seems good to merge, great work.

**Pull request merged**